newPost.sh

```diff
@@ -0,0 +1,46 @@
+read -p "Title: " title
+read -p "Date [$(date +%Y-%m-%d)]: " date
+read -p "Categories (space-separated): " categories
+read -p "Description: " description
+read -p "Post Image URL (full size): " img
+read -p "Post Image URL (smaller): " img_sm
+read -p "Author Name: " author_name
+read -p "Author Image URL: " author_img
+read -p "Author Bio: " author_bio
+read -p "Author E-Mail: " author_email
+read -p "Author GitHub URL: " author_github
+read -p "Author LinkedIn URL: " author_linkedin
+
+if [ -z "$date" ] # Default to current date
+then
+    date=$(date +%Y-%m-%d)
+fi
+
+title_url=$(echo "$title" | tr " " "-" | tr '[:upper:]' '[:lower:]')
+
+function add {
+    printf -- "$1\n" >> "_posts/$date-$title_url".md
+}
+
+# Append to file
+add "---"
+add "layout: post"
+add "title: $title"
+add "date: $date"
+add "categories:"
+for category in $categories
+do
+    add "\t- $category"
```

Comment:
You might want to list out what the categories available are (if there will be a preset list), or write a regex to run against grep in order to find what categories already are in-use.